chainstate/test-suite/src/tests/fungible_tokens_v1.rs

```diff
@@ -0,0 +1,2141 @@
+// and burned without unminting.
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn burn_from_total_supply_account(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut rng = make_seedable_rng(seed);
+        let mut tf = TestFramework::builder(&mut rng).build();
+
+        let token_min_supply_change_fee =
+            tf.chainstate.get_chain_config().token_min_supply_change_fee();
+
+        let amount_to_mint = Amount::from_atoms(rng.gen_range(2..1_000_000));
+        let amount_to_unmint = Amount::from_atoms(rng.gen_range(1..amount_to_mint.into_atoms()));
+
+        let (token_id, _, utxo_with_change) =
+            issue_token_from_genesis(&mut rng, &mut tf, TokenTotalSupply::Lockable);
+
+        let best_block_id = tf.best_block_id();
+        let (_, mint_tx_id) = mint_tokens_in_block(
+            &mut tf,
+            best_block_id,
+            utxo_with_change,
+            token_id,
+            amount_to_mint,
+            true,
+        );
+
+        // Spending from TokenTotalSupply and burning is not an error
+        // because it's basically minting new tokens without outputs (that can be scooped by miner)
+        // and burning the tokens from inputs without unminting.
+        tf.make_block_builder()
+            .add_transaction(
+                TransactionBuilder::new()
+                    .add_input(
+                        TxInput::from_account(
+                            AccountNonce::new(1),
+                            AccountSpending::TokenTotalSupply(token_id, amount_to_unmint),
+                        ),
+                        InputWitness::NoSignature(None),
+                    )
+                    .add_input(
+                        TxInput::from_utxo(mint_tx_id.into(), 1),
+                        InputWitness::NoSignature(None),
+                    )
+                    .add_input(
+                        TxInput::from_utxo(mint_tx_id.into(), 2),
+                        InputWitness::NoSignature(None),
+                    )
+                    .add_output(TxOutput::Burn(OutputValue::Coin(
+                        token_min_supply_change_fee,
+                    )))
+                    .add_output(TxOutput::Burn(OutputValue::TokenV1(
+                        token_id,
+                        amount_to_unmint,
+                    )))
+                    .build(),
+            )
+            .build_and_process()
+            .unwrap();
+
+        let circulating_supply =
+            TokensAccountingStorageRead::get_circulating_supply(&tf.storage, &token_id).unwrap();
+        assert_eq!(
+            circulating_supply,
+            Some((amount_to_mint + amount_to_unmint).unwrap())
+        );
+    });
+}
+
+// Issue a token, mint some and try to burn tokens and spend from lock supply account.
+// Check that it's not an error an that the supply actually is locked
+// and tokens are burned without unminting.
```

Comment:
Should this be an error? I may be wrong... I'm thinking, if someone signs an input to unmint, shouldn't be an error that they're trying to unmint, despite the supply being locked?

Reply:
There is no input to unmint only to lock supply. And that's what user got - locked supply and burned tokens without unminting.